### PR TITLE
[Rogue] Enable auto reforging feature + Small Subtlety APL tuning

### PR DESCRIPTION
**Pull request by TheBackstabi**
Title

### CODE REVIEW

**Review by 1337LutZ — APPROVED** [review]

File: ui/rogue/assassination/sim.ts [hunk 1]
```diff
@@ -54,7 +57,28 @@ const SPEC_CONFIG = registerSpecConfig(Spec.SpecAssassinationRogue, {
...
+		statCaps: (() => {
+			const expCap = new Stats().withStat(Stat.StatExpertise, 6.5 * 4 * Mechanics.EXPERTISE_PER_QUARTER_PERCENT_REDUCTION);
+			return expCap;
+		})(),
+		softCapBreakpoints: (() => {
+			const spellHitSoftCapConfig = {
+				stat: Stat.StatSpellHit,
+				breakpoints: [17 * Mechanics.SPELL_HIT_RATING_PER_HIT_CHANCE],
+				capType: StatCapType.TypeSoftCap,
+				postCapEPs: [0],
```
Comment (rosenrusinov) [inline ui/rogue/assassination/sim.ts:70]: Why is this a Soft cap and not a hard cap if the postCapEps is 0? That basically means it should be a hard cap if the value of the stat is zero after reaching this

File: ui/rogue/assassination/sim.ts [hunk 2]
```diff
@@ -54,7 +57,28 @@ const SPEC_CONFIG = registerSpecConfig(Spec.SpecAssassinationRogue, {
...
+			};
 
+			const meleeHitSoftCapConfig = {
+				stat: Stat.StatMeleeHit,
+				breakpoints: [8 * Mechanics.MELEE_HIT_RATING_PER_HIT_CHANCE, 27 * Mechanics.MELEE_HIT_RATING_PER_HIT_CHANCE],
+				capType: StatCapType.TypeSoftCap,
+				postCapEPs: [0.7, 0],
```
Comment (rosenrusinov) [inline ui/rogue/assassination/sim.ts:77]: Same here. Shouldnt the 27 be a hard cap and not a soft cap if the value after it becomes 0

Comment (1337LutZ) [reply ui/rogue/assassination/sim.ts:77]: I am not 100% sure if the reforge will actually work with a combination of having a hardCap + a softcap before the hardcap value 🤔

**Comment by 1337LutZ** [comment]
@rosenrusinov @TheBackstabi Just validated that the Reforge output is the same when changing it to this (and probably a bit more clear)
```
// Stat caps for reforge optimizer
statCaps: (() => {
	const spellHitCap = new Stats().withStat(Stat.StatSpellHit, 17 * Mechanics.SPELL_HIT_RATING_PER_HIT_CHANCE);
	const meleeHitCap = new Stats().withStat(Stat.StatMeleeHit, 8 * Mechanics.MELEE_HIT_RATING_PER_HIT_CHANCE);
	const expCap = new Stats().withStat(Stat.StatExpertise, 6.5 * 4 * Mechanics.EXPERTISE_PER_QUARTER_PERCENT_REDUCTION);
	return meleeHitCap.add(spellHitCap).add(expCap);
})(),
softCapBreakpoints: (() => {
	const meleeHitSoftCapConfig = {
		stat: Stat.StatMeleeHit,
		breakpoints: [8 * Mechanics.MELEE_HIT_RATING_PER_HIT_CHANCE],
		capType: StatCapType.TypeSoftCap,
		postCapEPs: [0.7],
	};

	return [meleeHitSoftCapConfig];
})(),
```

**Comment by rosenrusinov** [comment]
@1337LutZ The melee hard cap should be 27 tho, the soft cap is 8

**Pull request merged**